examples/js/controls/OrbitControls.js

```diff
@@ -312,11 +312,14 @@ THREE.OrbitControls = function ( object, domElement ) {
 
 	var panUp = function () {
 
-		var v = new THREE.Vector3();
+		var v = new THREE.Vector3( 1, 0, 0 );
 
 		return function panUp( distance, objectMatrix ) {
 
-			v.setFromMatrixColumn( objectMatrix, 1 ); // get Y column of objectMatrix
+			v.setFromMatrixColumn( objectMatrix, 0 ); // get X column of objectMatrix
+
+			v.applyAxisAngle( new THREE.Vector3( 0, 0, 1 ), Math.PI / 2 );
```

Comment:
This only works in your application because you have changed `camera.up`.

Also, avoid calling `new` unnecessarily. You are instantiating a `Vector3` that must be garbage-collected.

Yes, we can  improve `OrbitControls` panning, but this is not the way to do it.

Reply:
Ok. Copy.
I'll change my code.
Do you agree with the behaviour of the camera ?